autotx/agents/DelegateResearchTokensAgent.py

```diff
@@ -0,0 +1,138 @@
+    research: {{ "tasks": "Find the top 6 AI, GameFI and DEX tokens. It needs to be exactly 6 tokens total and without duplicates" }}
+
+    More examples:
+    research: {{ "tasks": "What's the 24 hours price change of SHIB?" }}
+    research: {{ "tasks": "Retrieve detailed information on the Ethereum token DAI including current price, market cap, and price change percentages." }}
+    research: {{ "tasks": "Find all tokens available in the DeFi category and filter them based on their performance on the Optimism network." }}
+    research: {{ "tasks": "Compare the price change percentages over the last 24 hours, 7 days, and 30 days for tokens within the Gaming category." }}
+    research: {{ "tasks": "Retrieve a list of all available token categories and then research specific details for the first five tokens in the NFT category sorted by volume in descending order." }}
+    research: {{ "tasks": "Get the Ethereum addresses for all tokens in the Yield Farming category on the Polygon network to prepare for potential trades." }}
+
+    Bad examples:
+    research: {{ "tasks": "Find best crpyto investment strategy" }}
+    research: {{ "tasks": "Research the best purchasing strategy" }}
+    research: {{ "tasks": "Provide financial advice on which tokens are likely to appreciate in value over the next year." }}
+    research: {{ "tasks": "Retrieve the transaction history for a specific Ethereum wallet address." }}
+    research: {{ "tasks": "Automatically monitor and alert the user about price changes for a specific token." }}
+    research: {{ "tasks": "Analyze the profitability of mining Ethereum tokens." }}
+
+    IMPORTANT: When querying for a number of tokens accross multiple categories, make sure to specify the number of tokens and the categories and that there should be no duplicates.
+    You MUST call the research tool once and pass in multiple tasks if needed.
+    """
+)
+
+description = dedent(
+    f"""
+    {name} is an AI assistant that's an expert in Ethereum tokens. 
+    The agent can also research, discuss, plan actions and advise the user.
+    """
+)
+
+def aggregate_chat_responses(chat: ChatResult) -> str:
+    summary = ""
+    # Go through the chat history in reverse order while ignoring tool messages and first message
+    for i in range(len(chat.chat_history) - 1, 0, -1):
+        last_message = chat.chat_history[i]
+        
+        if last_message["content"] and last_message["role"] != "tool":
+            summary = last_message["content"] + "\n" + summary
+
+    return summary.replace("TERMINATE", "")
+
+class ResearchUserQuery(AutoTxTool):
+    name: str = "research"
+    description: str = "Research user task"
+
+    def build_tool(self, autotx: AutoTx) -> Callable[[str], str]:
+        def run(
+            tasks: Annotated[str, "User tasks to research"]
+        ) -> str:
+            print(f"Researching user task:", tasks)
+
+            user_proxy_agent = UserProxyAgent(
```

Comment:
now that we have this new user proxy agent, which takes care of the research, would it make sense to clean up a bit the system message of the OG `user_proxy`?

Reply:
Maybe, but definitely wouldn't want to do it in this PR. Would require a lot of testing for regressions and at best we'd maybe be able to shave off a few sentences